modules/ip6/control/address.c

```diff
@@ -90,8 +90,8 @@ static int ip6_mcast_addr_add(struct iface *iface, const struct rte_ipv6_addr *i
 	if (i == ARRAY_DIM(maddrs->nh))
 		return errno_set(ENOSPC);
 
-	if ((nh = ip6_nexthop_lookup(iface->vrf_id, ip)) == NULL) {
-		if ((nh = ip6_nexthop_new(iface->vrf_id, GR_IFACE_ID_UNDEF, ip)) == NULL)
+	if ((nh = ip6_nexthop_lookup(iface->vrf_id, iface->id, ip)) == NULL) {
+		if ((nh = ip6_nexthop_new(iface->vrf_id, iface->id, ip)) == NULL)
```

Comment:
I am having second thoughts about scoping multicast addresses. Is this required?